stan/math/prim/arr/fun/array_builder.hpp

```diff
@@ -0,0 +1,36 @@
+#ifndef STAN_MATH_PRIM_ARR_FUN_ARRAY_BUILDER_HPP
+#define STAN_MATH_PRIM_ARR_FUN_ARRAY_BUILDER_HPP
+
+#include <stan/math/prim/arr/fun/promoter.hpp>
+#include <vector>
+
+namespace stan {
+  namespace math {
+
+    /**
+     * Structure for building up arrays in an expression (rather than
+     * in statements) using an argumentchaining add() method and
+     * a getter method array() to return the result.
+     *
+     * @tparam T1 type of arg1
+     */
+    template <typename T>
+    struct array_builder {
+      std::vector<T> x_;
+      array_builder() : x_() { }
+      template <typename F>
```

Comment:
Doc on all public functions, methods and constructors.  I think space between definitions would help rility.eadab